src/OpenLoco/src/Ui/Windows/Construction/Common.cpp

```diff
@@ -520,10 +520,31 @@ namespace OpenLoco::Ui::Windows::Construction
     }
 
     // 0x004A6E2B
+    // Update available road and rail for player company
     void updateAvailableRoadAndRailOptions()
     {
-        // update available road and rail for player company
-        call(0x004A6E2B);
+        if (getGameState().lastRoadOption == 0xFF)
+        {
+            uint8_t lastRoadOption = getGameState().lastTrackTypeOption;
+            if (lastRoadOption == 0xFF)
+            {
+                const auto availableObjects = companyGetAvailableRoads(CompanyManager::getControllingId());
+                lastRoadOption = availableObjects[0];
```

Comment:
just thinking about this again unfortunately as we now use a vector without a 0xFF sentinal we need to set this only if availableObjects is not empty.